chia/wallet/wallet_state_manager.py

```diff
@@ -1064,6 +1064,8 @@ async def new_coin_state(
                         ):
                             continue
 
+                    if coin_state.spent_height is not None and coin_name in trade_removals:
+                        trade_coin_removed.append(coin_state)
```

Comment:
What's the purpose of moving this?

Reply:
because it never reaches this code in the previous position because we skip it as we cannot find a wallet to store it to